cuda_core/cuda/core/experimental/_program.py

```diff
@@ -6,6 +6,157 @@
+        if self.device_c is not None:
+            self.formatted_options.append("--device-c".encode())
+        if self.device_w is not None:
+            self.formatted_options.append("--device-w".encode())
+        if self.relocatable_device_code is not None:
+            self.formatted_options.append(f"--relocatable-device-code={'true' if self.relocatable_device_code else 'false'}".encode())
+        if self.extensible_whole_program is not None:
+            self.formatted_options.append("--extensible-whole-program".encode())
+        if self.device_debug is not None:
+            self.formatted_options.append("--device-debug".encode())
+        if self.generate_line_info is not None:
+            self.formatted_options.append("--generate-line-info".encode())
+        if self.dopt is not None:
+            self.formatted_options.append(f"--dopt={'on' if self.dopt else 'off'}".encode())
+        if self.ptxas_options is not None:
+            self.formatted_options.append(f"--ptxas-options={self.ptxas_options}".encode())
+        if self.maxrregcount is not None:
+            self.formatted_options.append(f"--maxrregcount={self.maxrregcount}".encode())
+        if self.ftz is not None:
+            self.formatted_options.append(f"--ftz={'true' if self.ftz else 'false'}".encode())
+        if self.prec_sqrt is not None:
+            self.formatted_options.append(f"--prec-sqrt={'true' if self.prec_sqrt else 'false'}".encode())
+        if self.prec_div is not None:
+            self.formatted_options.append(f"--prec-div={'true' if self.prec_div else 'false'}".encode())
+        if self.fmad is not None:
+            self.formatted_options.append(f"--fmad={'true' if self.fmad else 'false'}".encode())
+        if self.use_fast_math is not None:
+            self.formatted_options.append("--use_fast_math".encode())
+        if self.extra_device_vectorization is not None:
+            self.formatted_options.append("--extra-device-vectorization".encode())
+        if self.modify_stack_limit is not None:
+            self.formatted_options.append(f"--modify-stack-limit={'true' if self.modify_stack_limit else 'false'}".encode())
+        if self.dlink_time_opt is not None:
+            self.formatted_options.append("--dlink-time-opt".encode())
+        if self.gen_opt_lto is not None:
+            self.formatted_options.append("--gen-opt-lto".encode())
+        if self.optix_ir is not None:
+            self.formatted_options.append("--optix-ir".encode())
+        if self.jump_table_density is not None:
+            self.formatted_options.append(f"--jump-table-density={self.jump_table_density}".encode())
+        if self.device_stack_protector is not None:
+            self.formatted_options.append(f"--device-stack-protector={'true' if self.device_stack_protector else 'false'}".encode())
+        if self.define_macro is not None:
+            if isinstance(self.define_macro, tuple):
```

Comment:
Do you know for sure that if `self.define_macro` is a tuple, it will have at least 2 items in it?  Also, if it's a tuple of 1 element, would the `else` clause be appropriate?

Reply:
I should add an assert and document it better in the options class.